core/src/main/java/org/apache/carbondata/core/scan/scanner/impl/BlockletFilterScanner.java

```diff
@@ -169,6 +169,10 @@ private BlockletScannedResult executeFilter(RawBlockletColumnChunks rawBlockletC
     // apply filter on actual data, for each page
     BitSetGroup bitSetGroup = this.filterExecuter.applyFilter(rawBlockletColumnChunks,
         useBitSetPipeLine);
+    // if bitSetGroup is nul, then new BitSetGroup object, which can avoid NPE
```

Comment:
Is `Lucene` will introduce this problem as you described or is it a bug caused by other scenario?

Reply:
It will throw exception when use lucent with search mode. I didn't check other scenario.

Reply:
Please describe why it is null, and when it is null

Reply:
before PR2242, it will be null if the bitSetGroup of rawBlockletColumnChunks is null, so we can remove this now